docs/multi-round.md

```diff
@@ -62,11 +62,20 @@ A processor instance is created only once, which means you can store information
 
 All KSP symbols will not be reusable across multiple rounds, as the resolution result can potentially change based on what was generated in a previous round. However, since KSP does not allow modifying existing code, some information such as the string value for a symbol name should still be reusable. To summarize, processors can store information from previous rounds but need to bear in mind that this information might be invalid in future rounds.
 
-#### Error Handling
-When an error occurs (defined by processor calling `KSPLogger.error()`), processing stops after the current round completes. All processors will call the `onError()` method and will **not** call the `finish()` method.
+#### Error and Exception Handling
+When an error(defined by processor calling `KSPLogger.error()`) or exception occurs, processing stops after the current round completes. All processors will call the `onError()` method and will **not** call the `finish()` method.
```

Comment:
One corner case I thought of looking at this: If `finish()` throws an exception or logs an error, what is the expected behavior?

Will `onError()` be called after `finish()` throws or logs an error?

Looking at the proposed implementation, it looks like

```kotlin
incrementalContext.updateCachesAndOutputs(dirtyFiles, codeGenerator.outputs, codeGenerator.sourceToOutputs)
```

will still be called if `finish()` throws or logs an error.

Reply:
Good catch. Currently if `finish` throws an exception, `onError` will still be called. However, regardless of whether we want this behavior or not, incremental data should not be updated upon errors.